nautobot/extras/tests/test_api.py

```diff
@@ -2236,6 +2238,83 @@ def test_dry_run_not_supported(self):
         self.assertHttpStatus(response, status.HTTP_405_METHOD_NOT_ALLOWED)
 
 
+class LogEntryTest(
+    APIViewTestCases.GetObjectViewTestCase,
+    APIViewTestCases.ListObjectsViewTestCase,
+):
+    """
+    Django core Admin app LogEntry records. We do not support user creation of these objects.
+    """
+
+    model = LogEntry
+    choices_fields = ["action_flag"]
+    filterset = LogEntryFilterSet
+    serializer_class = LogEntrySerializer
+
+    def setUp(self):
+        super().setUp()
+        # Only staff can see the logs.
+        self.user.is_staff = True
+        self.user.save()
+
+        admin_user = User.objects.create(username="admin", is_staff=True)
+        content_type = ContentType.objects.get_for_model(User)
+
+        self.log_entries = []
+        for action_flag, _ in ACTION_FLAG_CHOICES:
+            entry = LogEntry.objects.log_action(
+                user_id=admin_user.pk,
+                content_type_id=content_type.pk,
+                object_id=self.user.pk,
+                object_repr=str(self.user),
+                action_flag=action_flag,
+            )
+            self.log_entries.append(entry)
+
+    def _get_detail_url(self, instance):
+        return reverse("extras-api:logentry-detail", kwargs={"pk": instance.pk})
+
+    def test_get_object(self):
+        instance1 = self._get_queryset().first()
+        # Try GET to permitted object
+        url = self._get_detail_url(instance1)
+        response = self.client.get(url, **self.header)
+        self.assertHttpStatus(response, status.HTTP_200_OK)
+        self.assertIsInstance(response.data, dict)
+        # Fields that should be present in *ALL* model serializers:
+        self.assertIn("id", response.data)
+        self.assertEqual(str(response.data["id"]), str(instance1.pk))  # coerce to str to handle both int and uuid
+        self.assertIn("url", response.data)
+        self.assertIn("display", response.data)
+        self.assertIn("natural_slug", response.data)
+        self.assertIsInstance(response.data["display"], str)
+
+    def test_list_objects_csv(self):
+        """Disabled."""
```

Comment:
Not supported?

Reply:
Didn't think it was a good idea but open to suggestions.

Reply:
I didn't see anything in the code that would disable CSV export via REST - just the test case here. I would imagine that with the current implementation any admin user would be able to do CSV export just by passing `?format=csv` to the REST call.

Reply:
It's because of `is_staff=True` or not permissions, it fails:

```
======================================================================
FAIL: test_list_objects_csv (nautobot.extras.tests.test_api.LogEntryTest)
GET a list of objects in CSV format as an authenticated user with permission to view some objects.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/jathan/Library/Caches/pypoetry/virtualenvs/nautobot-b2ttWva6-py3.10/lib/python3.10/site-packages/django/test/utils.py", line 387, in inner
    return func(*args, **kwargs)
  File "/Users/jathan/sandbox/src/nautobot/nautobot/core/testing/api.py", line 566, in test_list_objects_csv
    self.assertEqual(1, len(rows))
AssertionError: 1 != 2

----------------------------------------------------------------------
```

Need to think this one through.